src/hotspot/share/opto/superword.cpp

```diff
@@ -2141,7 +2081,7 @@ class PacksetGraph {
 
   // Create nodes (from packs and scalar-nodes), and add edges, based on the dependency graph.
   void build() {
-    const GrowableArray<Node_List*>& packset = _slp->packset();
+    const PackSet& packset = _slp->packset();
```

Comment:
You are right, I could actually remove the SuperWord reference from the `PacksetGraph` completely, and just pass the components, as `const` references. That would be really nice.

If it is ok for you, I will do this in a future RFE.

Actually, I plan to completely overhaul the `PacksetGraph`. It will be transformed to the `VTransformGraph`, and it will do:
- Cycle checking (like today)
- Evaluate the cost-model
- Execute: each node knows how to replace its packed scalar nodes with vector nodes (basically refactoring `SuperWord::output` away)
- etc.

We may even be able to take the `VTransformGraph` and try to widen all nodes, or make transformations on this graph, a simplified version of IGVN. I have lots of ideas that would be unlocked with this new graph-based approach.

Reply:
Sounds exciting! 

> If it is ok for you, I will do this in a future RFE.

Sure, let's leave this as it is now.